stage0/src/apic.rs

```diff
@@ -19,6 +19,36 @@ use core::arch::x86_64::__cpuid;
 use bitflags::bitflags;
 use x86_64::{registers::model_specific::Msr, PhysAddr};
 
+pub(crate) enum ApicRegister {
+    X2apic(Msr),
+}
+
+impl ApicRegister {
+    fn read(&self) -> u64 {
+        match self {
+            ApicRegister::X2apic(msr) => unsafe { msr.read() },
```

Comment:
nit: adding some safety comments would be nice. Also, I plan on configuring clippy::undocumented_unsafe_blocks at some point, which would force us to document all unsafe blocks.

Reply:
Yep, I agree, I got lazy here.

However, in my furter work I've ended up refactoring a lot of this code anyway, so I'll make sure it has proper comments there and let this slide for now.

Having the Clippy check would be awesome.